### PR TITLE
[WIP] feat(Core/Account): RBAC

### PR DESCRIPTION
**Pull request by IntelligentQuantum**
## Changes Proposed:
-  
-  

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/5704

## Tests Performed:

- 
- 

## How to Test the Changes:

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Kitzunu** [comment]
This is cherry-picked from TC right? If so I will add all the co-authors that workdd on TC for it

**Comment by IntelligentQuantum** [comment]
> This is cherry-picked from TC right? If so I will add all the co-authors that workdd on TC for it

Yeah

**Comment by Winfidonarleyan** [comment]
I'm not sure, but maybe it would be better to make a separate manager for all this?

File: src/server/game/Server/WorldSession.h [hunk 1]
```diff
@@ -1083,6 +1094,7 @@ class WorldSession
     uint32 recruiterId;
     bool isRecruiter;
     LockedQueue<WorldPacket*> _recvQueue;
+    rbac::RBACData* _RBACData;
```
Comment (Winfidonarleyan) [inline src/server/game/Server/WorldSession.h:1097]: Shared ptr maybe?

**Comment by IntelligentQuantum** [comment]
@azerothcore/testers Can someone test this pr please.

**Comment by claudiodfc** [comment]
@IntelligentQuantum Bro how do I test this?

File: src/server/game/World/World.cpp [hunk 2]
```diff
@@ -1392,6 +1392,9 @@ void World::LoadConfigSettings(bool reload)
 
     m_int_configs[CONFIG_PACKET_SPOOF_BANDURATION] = sConfigMgr->GetOption<int32>("PacketSpoof.BanDuration", 86400);
 
+    // Accountpassword Secruity
+    m_int_configs[CONFIG_ACC_PASSCHANGESEC] = sConfigMgr->GetIntDefault("Account.PasswordChangeSecurity", 0);
```
Comment (Winfidonarleyan) [inline src/server/game/World/World.cpp:1396]: This need?

Comment (IntelligentQuantum) [reply src/server/game/World/World.cpp:1396]: yes

**Comment by Kitzunu** [comment]
Status?

**Pull request closed without merging**